python/ray/serve/pipeline/api.py

```diff
@@ -53,9 +56,42 @@ def build(ray_dag_root_node: DAGNode):
         Assuming we have non-JSON serializable or inline defined class or
         function in local pipeline development.
 
-        >>> deployments = serve.pipeline.build(ray_dag) # it can be method node
-        >>> deployments = serve.pipeline.build(m1) # or just a regular node.
+        >>> deployments = serve.build(ray_dag) # it can be method node
+        >>> deployments = serve.build(m1) # or just a regular node.
     """
     serve_root_dag = ray_dag_root_node.apply_recursive(transform_ray_dag_to_serve_dag)
     deployments = extract_deployments_from_serve_dag(serve_root_dag)
-    return deployments
+    deployments_with_http = mark_exposed_deployment_in_serve_dag(deployments)
+
+    return deployments_with_http
+
+
+def get_and_validate_exposed_deployment(deployments: List[Deployment]) -> Deployment:
+    """Validation for http route prefixes for a list of deployments in pipeline.
+
+    Ensures:
+        1) One and only one exposed deployment with given route prefix.
+        2) All other not exposed deployments should have prefix of None.
+        3) Exposed deployment should only have given route prefix, not others.
+    """
+
+    exposed_deployments = []
+    for deployment in deployments:
+        if deployment.route_prefix == "/":
+            exposed_deployments.append(deployment)
+        # Condition 2) and 3)
+        elif deployment.route_prefix is not None:
+            raise ValueError(
+                "Exposed deployment should not have route prefix other than "
+                f"'/', found: {deployment.route_prefix}"
+            )
```

Comment:
sorry i should have caught this earlier. We should support user providing their own exposed deployment prefix. for example `Driver.options(route_prefix="/api/v1")`

Reply:
cool i think the only change needed on this PR is to remove this assertion on "/" so user can bring their own and reconfigure it later ?

Reply:
yes! and tests